### PR TITLE
refactor: split commands logic

### PR DESCRIPTION
**Pull request by cristianoliveira**
I've done some refactor to allow add new commands easier. 
I abstracted the param extraction from request. I think it makes the server's code cleaner.

### CODE REVIEW

File: src/bojack/commands/command.cr [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require "../memory"
+
+module Bojack
+  module Commands
+    # Represents a Bojack command
+    #
+    # It is a interface that each command
```
Comment (mauricioabreu) [inline src/bojack/commands/command.cr]: Crystal supports `abstract` types. It this the case where you are enforcing users to have the `execute` method defined?
Reference: https://crystal-lang.org/docs/syntax_and_semantics/virtual_and_abstract_types.html

Comment (cristianoliveira) [inline src/bojack/commands/command.cr]: Great! Let me change it. It is exactly what I tried to represent with this class.

Comment (mauricioabreu) [inline src/bojack/commands/command.cr]: Cool! Nice code.

**Comment by marceloboeira** [comment]
@cristianoliveira amazing, thanks a lot. Since I'm without my computer and you have write permission can your rebase this with master and merge without the merge commit ?

Thanks!

Only thing, take a look at te contributors guide for commit messages, however for now dont worry about it :)

File: src/bojack/commands/command.cr (hunk 1 above)
Comment (marceloboeira) [inline src/bojack/commands/command.cr]: @mauricioabreu well noticed my friend!

**Comment by cristianoliveira** [comment]
Done!

Sorry, I just notice the wrong messages after the third commit :sweat_smile:

File: src/bojack/commands/command.cr (hunk 1 above)
Comment (mauricioabreu) [inline src/bojack/commands/command.cr]: You are welcome @marceloboeira

**Pull request merged**